{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "\n",
    "import sys\n",
    "import argparse\n",
    "import sys\n",
    "import re\n",
    "import os \n",
    "from pysam import VariantFile\n",
    "\n",
    "def vcf2DP(vcf_file_name,distal_test):\n",
    "    VCFin = VariantFile(vcf_file_name)\n",
    "    big_info = []\n",
    "    for rec in VCFin.fetch(distal_test[0],int(distal_test[1]), int(distal_test[2])):\n",
    "        #print (rec)\n",
    "        #print (rec.samples.keys())\n",
    "         \n",
    "        #print (rec.pos)\n",
    "        #print (rec.contig)\n",
    "        #print (rec.id)\n",
    "        #print (rec.alleles)\n",
    "        #combined_info.append(rec.contig)\n",
    "        #combined_info.append(rec.pos)\n",
    "        #combined_info.append(rec.id)\n",
    "        #combined_info.append(rec.alleles)\n",
    "        #print(len(rec.alleles[0]))\n",
    "        #print(len(rec.alleles[1]))\n",
    "        combined_info =[]\n",
    "        if len(rec.alleles[0]) ==1 and  len(rec.alleles[1]) ==1:\n",
    "            \n",
    "            combined_info.append(rec.contig)\n",
    "            combined_info.append(rec.pos)\n",
    "            combined_info.append(rec.id)\n",
    "            combined_info.append(rec.alleles)\n",
    "            combined_info.append(rec.filter.keys())\n",
    "            for test in rec.samples.values():\n",
    "            #print (test[\"GT\"])\n",
    "                combined_info.append(test[\"GT\"])\n",
    "            #print (test[\"AD\"])\n",
    "                combined_info.append(test[\"AD\"])\n",
    "            #print (test[\"DP\"])\n",
    "                combined_info.append(test[\"DP\"])\n",
    "            #print (test[\"GQ\"])\n",
    "            #print (test[\"PL\"])\n",
    "            #print (\"**\")\n",
    "        big_info.append(combined_info)\n",
    "    \n",
    "    big_info_corrected = [x for x in big_info if x != []]\n",
    "    return(big_info_corrected)\n",
    "\n",
    "\n",
    "path = \"/Volumes/Work_drive/prj/BAV_TAV/data/raw_internal/Interaction_callsFeb/SNVs/recalibrated_SNPs\"\n",
    "os.chdir(path)\n",
    "\n",
    "\n",
    "#bedFiles = \"sorted_uniq_PD_Interaction.bed\"\n",
    "\n",
    "bedFiles = \"BAV_PD.bed\"\n",
    "snp_file = \"SNV_PD.txt\"\n",
    "\n",
    "files = [\n",
    "        \"TAV2431_snp_indel.recalibrated.vcf.gz\", \n",
    "        \"TAV2515_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"TAV2709_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"BAV2375_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"BAV2424_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"BAV2714_snp_indel.recalibrated.vcf.gz\"\n",
    "        ]\n",
    "\n",
    "f_r = open (bedFiles, \"r\")\n",
    "f_w = open (snp_file, \"w\")\n",
    "\n",
    "for interactors in f_r:\n",
    "    \n",
    "    print(interactors)\n",
    "    line = interactors.strip().split(\"\\t\")\n",
    "    distal_element = (line[0], int(line[1]), int(line[2]))\n",
    "    #print (distal_element)\n",
    "    \n",
    "    for file_name in files:\n",
    "        print (file_name)\n",
    "        combined = vcf2DP(file_name, distal_element)\n",
    "        if not combined:\n",
    "            f_w.write(\"\\t\".join(line)+\"\\t\"+file_name+\"\\t\"+\"NA\")\n",
    "            f_w.write(\"\\n\")\n",
    "            \n",
    "            \n",
    "        else:\n",
    "            #print (combined)\n",
    "            for i in combined:\n",
    "                \n",
    "                \n",
    "                allele_combo = \"/\".join(i[3])\n",
    "                genotype_combo = \"/\".join([str(x) for x in i[5]])\n",
    "                DP_combo = \"/\".join([str(x) for x in i[6]])\n",
    "                \n",
    "                if i[2] == None :\n",
    "                    all = i[0]+\":\"+str(i[1])+\":\"+\".\"+\":\"+allele_combo+\":\"+genotype_combo+\":\"+DP_combo+\":\"+str(i[7])\n",
    "                else:\n",
    "                     all = i[0]+\":\"+str(i[1])+\":\"+i[2]+\":\"+allele_combo+\":\"+genotype_combo+\":\"+DP_combo+\":\"+str(i[7])\n",
    "                print (all)\n",
    "            \n",
    "                f_w.write(\"\\t\".join(line)+\"\\t\"+file_name+\"\\t\"+all)\n",
    "                f_w.write(\"\\n\")\n",
    "            \n",
    "\n",
    "f_r.close()\n",
    "f_w.close()\n",
    "\n",
    "\n",
    "\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import argparse\n",
    "import sys\n",
    "import re\n",
    "import os \n",
    "from pysam import VariantFile\n",
    "\n",
    "def vcf2DP(vcf_file_name,distal_test):\n",
    "    VCFin = VariantFile(vcf_file_name)\n",
    "    big_info = []\n",
    "    for rec in VCFin.fetch(distal_test[0],int(distal_test[1]), int(distal_test[2])):\n",
    "        combined_info =[]\n",
    "        \n",
    "        if len(rec.alleles[0]) ==1 and  len(rec.alleles[1]) ==1:\n",
    "            for test in rec.samples.values():\n",
    "                if test[\"GT\"] == (0,1) or test[\"GT\"] == (1,1):\n",
    "                    #print (test[\"GT\"])\n",
    "                \n",
    "                    combined_info.append(rec.contig)\n",
    "                    combined_info.append(rec.pos)\n",
    "                    combined_info.append(rec.id)\n",
    "                    combined_info.append(rec.alleles)\n",
    "                    combined_info.append(rec.filter.keys())\n",
    "                    combined_info.append(test[\"GT\"])\n",
    "                    combined_info.append(test[\"AD\"])\n",
    "                    combined_info.append(test[\"DP\"])\n",
    "                \n",
    "        big_info.append(combined_info)\n",
    "    big_info_corrected = [x for x in big_info if x != []]\n",
    "    return(big_info_corrected)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [],
   "source": [
    "def vcf2DP(vcf_file_name,distal_test):\n",
    "    \n",
    "    VCFin = VariantFile(vcf_file_name)\n",
    "    \n",
    "    BAV1= []\n",
    "    BAV2= []\n",
    "    BAV3= []\n",
    "    TAV1= []\n",
    "    TAV2= []\n",
    "    TAV3= []\n",
    "    \n",
    "    for rec in VCFin.fetch(distal_test[0],int(distal_test[1]), int(distal_test[2])):\n",
    "        if len(rec.alleles[0]) ==1 and  len(rec.alleles[1]) ==1 and rec.filter.keys()[0] == \"PASS\":\n",
    "            combined_info =[]\n",
    "            \n",
    "            for test in rec.samples.values():    \n",
    "                AD_combo = \"/\".join([str(x) for x in test[\"AD\"]])\n",
    "                genotype_combo = \"/\".join([str(x) for x in test[\"GT\"]])\n",
    "                if genotype_combo == \"None/None\":\n",
    "                    genotype_combo = \"./.\"\n",
    "                all_test = rec.contig+\"_\"+str(rec.pos)+\"_\"+rec.id+\"_\"+\"|\".join(rec.alleles)+\"_\"+genotype_combo+\"_\"+AD_combo\n",
    "                combined_info.append(all_test)\n",
    "                \n",
    "            BAV1.append(combined_info[0])\n",
    "            BAV2.append(combined_info[1])\n",
    "            BAV3.append(combined_info[2])\n",
    "            TAV1.append(combined_info[3])\n",
    "            TAV2.append(combined_info[4])\n",
    "            TAV3.append(combined_info[5])\n",
    "    return (BAV1,BAV2,BAV3,TAV1,TAV2,TAV3)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [],
   "source": [
    "distal_test = [\"chr1\", 6565767,6573159]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(['chr1_6566715_rs9787174_C|T_0/0_5/0',\n",
       "  'chr1_6571924_rs4525068_A|G_1/1_0/11',\n",
       "  'chr1_6572079_rs4243828_T|C_1/1_0/47',\n",
       "  'chr1_6572175_rs3935435_G|A_0/0_45/0',\n",
       "  'chr1_6572271_rs9787374_C|G_0/0_38/0',\n",
       "  'chr1_6572332_rs4509598_A|G_1/1_0/20',\n",
       "  'chr1_6573038_rs10442706_C|G_1/1_0/22'],\n",
       " ['chr1_6566715_rs9787174_C|T_0/0_12/0',\n",
       "  'chr1_6571924_rs4525068_A|G_1/1_0/8',\n",
       "  'chr1_6572079_rs4243828_T|C_1/1_0/53',\n",
       "  'chr1_6572175_rs3935435_G|A_0/0_38/0',\n",
       "  'chr1_6572271_rs9787374_C|G_0/0_26/0',\n",
       "  'chr1_6572332_rs4509598_A|G_1/1_0/18',\n",
       "  'chr1_6573038_rs10442706_C|G_1/1_0/19'],\n",
       " ['chr1_6566715_rs9787174_C|T_1/1_0/7',\n",
       "  'chr1_6571924_rs4525068_A|G_1/1_0/4',\n",
       "  'chr1_6572079_rs4243828_T|C_1/1_0/15',\n",
       "  'chr1_6572175_rs3935435_G|A_0/1_7/16',\n",
       "  'chr1_6572271_rs9787374_C|G_0/1_6/3',\n",
       "  'chr1_6572332_rs4509598_A|G_1/1_0/3',\n",
       "  'chr1_6573038_rs10442706_C|G_0/1_1/3'],\n",
       " ['chr1_6566715_rs9787174_C|T_./._1/0',\n",
       "  'chr1_6571924_rs4525068_A|G_1/1_0/2',\n",
       "  'chr1_6572079_rs4243828_T|C_1/1_0/37',\n",
       "  'chr1_6572175_rs3935435_G|A_0/1_18/18',\n",
       "  'chr1_6572271_rs9787374_C|G_0/1_7/8',\n",
       "  'chr1_6572332_rs4509598_A|G_1/1_0/2',\n",
       "  'chr1_6573038_rs10442706_C|G_1/1_0/9'],\n",
       " ['chr1_6566715_rs9787174_C|T_0/1_4/3',\n",
       "  'chr1_6571924_rs4525068_A|G_0/0_1/0',\n",
       "  'chr1_6572079_rs4243828_T|C_1/1_0/32',\n",
       "  'chr1_6572175_rs3935435_G|A_0/1_21/22',\n",
       "  'chr1_6572271_rs9787374_C|G_0/1_11/10',\n",
       "  'chr1_6572332_rs4509598_A|G_1/1_0/4',\n",
       "  'chr1_6573038_rs10442706_C|G_0/1_11/2'],\n",
       " ['chr1_6566715_rs9787174_C|T_0/0_2/0',\n",
       "  'chr1_6571924_rs4525068_A|G_1/1_0/2',\n",
       "  'chr1_6572079_rs4243828_T|C_1/1_0/17',\n",
       "  'chr1_6572175_rs3935435_G|A_0/0_22/0',\n",
       "  'chr1_6572271_rs9787374_C|G_0/0_9/0',\n",
       "  'chr1_6572332_rs4509598_A|G_1/1_0/4',\n",
       "  'chr1_6573038_rs10442706_C|G_1/1_0/8'])"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = \"/Volumes/Work_drive/prj/BAV_TAV/data/raw_internal/SNP_calling/recalibrated_SNPs\"\n",
    "os.chdir(path)\n",
    "\n",
    "file ='raw_var_snp_indel.recalibrated.vcf.gz'\n",
    "\n",
    "vcf2DP(file, distal_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('chr1', 7817602, 7818563)\n"
     ]
    }
   ],
   "source": [
    "path = \"/Volumes/Work_drive/prj/BAV_TAV/data/raw_internal/SNP_calling/recalibrated_SNPs\"\n",
    "os.chdir(path)\n",
    "\n",
    "bedFiles = \"BAV_PD.bed\"\n",
    "f_r = open (bedFiles, \"r\")\n",
    "\n",
    "\n",
    "for interactors in f_r:\n",
    "    line = interactors.strip().split(\"\\t\")\n",
    "    distal_element = (line[0], int(line[1]), int(line[2]))    \n",
    "    print (distal_element)\n",
    "    break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1_6566715_rs9787174_C|T_0/0_5/0', 'chr1_6571924_rs4525068_A|G_1/1_0/11', 'chr1_6572079_rs4243828_T|C_1/1_0/47', 'chr1_6572175_rs3935435_G|A_0/0_45/0', 'chr1_6572271_rs9787374_C|G_0/0_38/0', 'chr1_6572332_rs4509598_A|G_1/1_0/20', 'chr1_6573038_rs10442706_C|G_1/1_0/22']\n",
      "['chr1_6566715_rs9787174_C|T_0/0_12/0', 'chr1_6571924_rs4525068_A|G_1/1_0/8', 'chr1_6572079_rs4243828_T|C_1/1_0/53', 'chr1_6572175_rs3935435_G|A_0/0_38/0', 'chr1_6572271_rs9787374_C|G_0/0_26/0', 'chr1_6572332_rs4509598_A|G_1/1_0/18', 'chr1_6573038_rs10442706_C|G_1/1_0/19']\n",
      "['chr1_6566715_rs9787174_C|T_1/1_0/7', 'chr1_6571924_rs4525068_A|G_1/1_0/4', 'chr1_6572079_rs4243828_T|C_1/1_0/15', 'chr1_6572175_rs3935435_G|A_0/1_7/16', 'chr1_6572271_rs9787374_C|G_0/1_6/3', 'chr1_6572332_rs4509598_A|G_1/1_0/3', 'chr1_6573038_rs10442706_C|G_0/1_1/3']\n",
      "['chr1_6566715_rs9787174_C|T_./._1/0', 'chr1_6571924_rs4525068_A|G_1/1_0/2', 'chr1_6572079_rs4243828_T|C_1/1_0/37', 'chr1_6572175_rs3935435_G|A_0/1_18/18', 'chr1_6572271_rs9787374_C|G_0/1_7/8', 'chr1_6572332_rs4509598_A|G_1/1_0/2', 'chr1_6573038_rs10442706_C|G_1/1_0/9']\n",
      "['chr1_6566715_rs9787174_C|T_0/1_4/3', 'chr1_6571924_rs4525068_A|G_0/0_1/0', 'chr1_6572079_rs4243828_T|C_1/1_0/32', 'chr1_6572175_rs3935435_G|A_0/1_21/22', 'chr1_6572271_rs9787374_C|G_0/1_11/10', 'chr1_6572332_rs4509598_A|G_1/1_0/4', 'chr1_6573038_rs10442706_C|G_0/1_11/2']\n",
      "['chr1_6566715_rs9787174_C|T_0/0_2/0', 'chr1_6571924_rs4525068_A|G_1/1_0/2', 'chr1_6572079_rs4243828_T|C_1/1_0/17', 'chr1_6572175_rs3935435_G|A_0/0_22/0', 'chr1_6572271_rs9787374_C|G_0/0_9/0', 'chr1_6572332_rs4509598_A|G_1/1_0/4', 'chr1_6573038_rs10442706_C|G_1/1_0/8']\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "import argparse\n",
    "import sys\n",
    "import re\n",
    "import os \n",
    "from pysam import VariantFile\n",
    "\n",
    "path = \"/Volumes/Work_drive/prj/BAV_TAV/data/raw_internal/SNP_calling/recalibrated_SNPs\"\n",
    "os.chdir(path)\n",
    "\n",
    "bedFiles = \"BAV_PD.bed\"\n",
    "f_r = open (bedFiles, \"r\")\n",
    "\n",
    "file ='raw_var_snp_indel.recalibrated.vcf.gz'\n",
    "\n",
    "#for interactors in f_r:\n",
    "#    line = interactors.strip().split(\"\\t\")\n",
    "#    distal_element = (line[0], int(line[1]), int(line[2]))    \n",
    "#    print (distal_element)\n",
    "\n",
    "\n",
    "VCFin = VariantFile(file)\n",
    "BAV1= []\n",
    "BAV2= []\n",
    "BAV3= []\n",
    "TAV1= []\n",
    "TAV2= []\n",
    "TAV3= []\n",
    "\n",
    "\n",
    "for rec in VCFin.fetch (\"chr1\",6565767,6573159):  \n",
    "    if len(rec.alleles[0]) ==1 and  len(rec.alleles[1]) ==1 and rec.filter.keys()[0] == \"PASS\":\n",
    "        \n",
    "        combined_info =[]\n",
    "        \n",
    "        for test in rec.samples.values():    \n",
    "            AD_combo = \"/\".join([str(x) for x in test[\"AD\"]])\n",
    "            genotype_combo = \"/\".join([str(x) for x in test[\"GT\"]])\n",
    "            if genotype_combo == \"None/None\":\n",
    "                genotype_combo = \"./.\"\n",
    "            all_test = rec.contig+\"_\"+str(rec.pos)+\"_\"+rec.id+\"_\"+\"|\".join(rec.alleles)+\"_\"+genotype_combo+\"_\"+AD_combo\n",
    "            combined_info.append(all_test) \n",
    "     \n",
    "        BAV1.append(combined_info[0])\n",
    "        BAV2.append(combined_info[1])\n",
    "        BAV3.append(combined_info[2])\n",
    "        TAV1.append(combined_info[3])\n",
    "        TAV2.append(combined_info[4])\n",
    "        TAV3.append(combined_info[5])\n",
    "    \n",
    "print (BAV1)\n",
    "print (BAV2)\n",
    "print (BAV3)\n",
    "print (TAV1)\n",
    "print (TAV2)\n",
    "print (TAV3)\n",
    "  \n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = \"/Volumes/Work_drive/prj/BAV_TAV/data/raw_internal/Interaction_callsFeb/SNVs/recalibrated_SNPs\"\n",
    "os.chdir(path)\n",
    "\n",
    "\n",
    "#bedFiles = \"sorted_uniq_PD_Interaction.bed\"\n",
    "\n",
    "bedFiles = \"BAV_PD.bed\"\n",
    "#snp_file = \"SNV_PD.txt\"\n",
    "\n",
    "files = [\n",
    "        \"TAV2431_snp_indel.recalibrated.vcf.gz\", \n",
    "        \"TAV2515_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"TAV2709_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"BAV2375_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"BAV2424_snp_indel.recalibrated.vcf.gz\",\n",
    "        \"BAV2714_snp_indel.recalibrated.vcf.gz\"\n",
    "        ]\n",
    "\n",
    "f_r = open (bedFiles, \"r\")\n",
    "#f_w = open (snp_file, \"w\")\n",
    "\n",
    "all_dict = {}\n",
    "for interactors in f_r:\n",
    "    line = interactors.strip().split(\"\\t\")\n",
    "    distal_element = (line[0], int(line[1]), int(line[2]))    \n",
    "    combo = all_dict.get(distal_element, [])\n",
    "    for file_name in files:\n",
    "        combo.append(file_name)\n",
    "        all_dict[distal_element]= combo \n",
    "f_r.close()\n",
    "\n",
    "\n",
    "combined_list= []\n",
    "for keys, values in all_dict.items():\n",
    "    all_al = []\n",
    "    all_al.append(keys)\n",
    "    for val in values:\n",
    "        #print (val)\n",
    "        combined = vcf2DP(val, keys)\n",
    "        #print (combined)\n",
    "        if not combined:\n",
    "            all_al.append(\"NA\") \n",
    "        else:\n",
    "            collector = []\n",
    "            for i in combined:\n",
    "                allele_combo = \"/\".join(i[3])\n",
    "                genotype_combo = \"/\".join([str(x) for x in i[5]])\n",
    "                DP_combo = \"/\".join([str(x) for x in i[6]])\n",
    "                if i[2] == None :\n",
    "                    all = i[0]+\":\"+str(i[1])+\":\"+\".\"+\":\"+allele_combo+\":\"+genotype_combo+\":\"+DP_combo+\":\"+str(i[7])\n",
    "                    collector.append(all)\n",
    "                else:\n",
    "                    all = i[0]+\":\"+str(i[1])+\":\"+i[2]+\":\"+allele_combo+\":\"+genotype_combo+\":\"+DP_combo+\":\"+str(i[7])\n",
    "                    collector.append(all)\n",
    "                #print (all)\n",
    "            all_al.append(collector)\n",
    "            \n",
    "    print(all_al)\n",
    "    \n",
    "    combined_list.append(all_al)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_file = \"SNV_PD_2.txt\"\n",
    "fh = open (snp_file, \"w\")\n",
    "\n",
    "for element in combined_list:\n",
    "    print(\"\\t\".join([str(x) for x in element[0]]))\n",
    "    print (\"|\".join(element[1]))\n",
    "    print (\"|\".join(element[2]))\n",
    "    print (\"|\".join(element[3]))\n",
    "    print (\"|\".join(element[4]))\n",
    "    print (\"|\".join(element[5]))\n",
    "    print (\"|\".join(element[6]))\n",
    "    final_act =\"\\t\".join([str(x) for x in element[0]])+\"\\t\"+\"|\".join(element[1])+\"\\t\"+\"|\".join(element[2])+\"\\t\"+\"|\".join(element[3])+\"\\t\"+\"|\".join(element[4])+\"\\t\"+\"|\".join(element[5])+\"\\t\"+\"|\".join(element[6])\n",
    "                \n",
    "    fh.write(final_act)\n",
    "    fh.write(\"\\n\")\n",
    "                           \n",
    "fh.close()\n",
    "               "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bgzip -c TAV2709_snp_indel.recalibrated.vcf > TAV2709_snp_indel.recalibrated.vcf.gz\n",
    "tabix -p vcf TAV2709_snp_indel.recalibrated.vcf.gz\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#path = \"/Volumes/Work_drive/prj/BAV_TAV/data/raw_internal/Interaction_callsFeb/SNVs/recalibrated_SNPs\"\n",
    "#os.chdir(path)\n",
    "\n",
    "#distal_test = (\"chr1\",7817602,7818563)\n",
    "#distal_test = (\"chr12\",122583734,122584790)\n",
    "#file_name = \"BAV2375_snp_indel.recalibrated.vcf.gz\"\n",
    "#file_name = \"BAV2714_snp_indel.recalibrated.vcf.gz\"\n",
    "#test = vcf2DP(file_name,distal_test)\n",
    "\n",
    "#test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "            \n",
    "                \n",
    "                \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
